libs/extensions/tabular/exec/src/lib/xlsx-interpreter-executor.ts

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { strict as assert } from 'assert';
+
+import * as R from '@jvalue/jayvee-execution';
+import {
+  AbstractBlockExecutor,
+  BinaryFile,
+  BlockExecutorClass,
+  ExecutionContext,
+  Workbook,
+  implementsStatic,
+} from '@jvalue/jayvee-execution';
+import { IOType } from '@jvalue/jayvee-language-server';
+import * as xlsx from 'xlsx';
+
+@implementsStatic<BlockExecutorClass>()
+export class XLSXInterpreterExecutor extends AbstractBlockExecutor<
+  IOType.FILE,
+  IOType.WORKBOOK
+> {
+  public static readonly type = 'XLSXInterpreter';
+
+  constructor() {
+    super(IOType.FILE, IOType.WORKBOOK);
+  }
+
+  async doExecute(
+    file: BinaryFile,
+    context: ExecutionContext,
+  ): Promise<R.Result<Workbook>> {
+    context.logger.logDebug(`reading from xlsx file`);
+    const workBookFromFile = xlsx.read(file.content, { dense: true });
+    const workbook = new Workbook();
+    for (const workSheetName of workBookFromFile.SheetNames) {
+      const workSheet = workBookFromFile.Sheets[workSheetName];
+      assert(workSheet !== undefined);
+
+      const workSheetDataArray = Array.prototype.map.call<
```

Comment:
To be quite frank, because I did't manage to do it differently eg calling `map` directly without failing lint and or type checks. Maybe I can have another look with @rhazn.

Reply:
Yes, please bring this up in our sync, we can do it live.